### PR TITLE
feat: service manager frontend view functions

### PR DESCRIPTION
**Pull request by stevennevins**

### CODE REVIEW

File: src/ServiceManagerBase.sol [hunk 1]
```diff
@@ -107,6 +107,39 @@ abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
...
+        return bytesArrayToUint32Array(BitmapUtils.bitmapToBytesArray(operatorQuorumBitmap));
+
+
+    }
+
+    function getAVSQuorums() external view returns (uint32[] memory){
+       uint192 quorumCount = _registryCoordinator.quorumCount();
+
+        uint192 bitmap;
+        for (uint192 i = 0; i < quorumCount; i++) {
```
Comment (gpsanant) [inline src/ServiceManagerBase.sol:122]: `1 << quorumCount - 1` should work?

File: src/ServiceManagerBase.sol [hunk 2]
```diff
@@ -107,6 +107,39 @@ abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
...
+            bitmap |= (uint192(1) << i);
+        }
+
+       return bytesArrayToUint32Array(BitmapUtils.bitmapToBytesArray(bitmap));
+
+    }
+
+    function getStrategiesForQuorum(uint8 quorumNumber) external view returns(address[] memory){
+        require(quorumNumber <= _registryCoordinator.quorumCount(), "Invalid Quorum");
```
Comment (gpsanant) [inline src/ServiceManagerBase.sol:131]: revert message style

File: src/ServiceManagerBase.sol [hunk 3]
```diff
@@ -107,6 +107,39 @@ abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
...
+
+    function getAVSQuorums() external view returns (uint32[] memory){
+       uint192 quorumCount = _registryCoordinator.quorumCount();
+
+        uint192 bitmap;
+        for (uint192 i = 0; i < quorumCount; i++) {
+            bitmap |= (uint192(1) << i);
+        }
+
+       return bytesArrayToUint32Array(BitmapUtils.bitmapToBytesArray(bitmap));
```
Comment (gpsanant) [inline src/ServiceManagerBase.sol:126]: u can edit BitMapUtils if that's cleaner

File: src/ServiceManagerBase.sol [hunk 4]
```diff
@@ -107,6 +107,39 @@ abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
...
+
+    }
+
+    function getStrategiesForQuorum(uint8 quorumNumber) external view returns(address[] memory){
```
Comment (gpsanant) [inline src/ServiceManagerBase.sol:130]: should the param be a uint32?

File: src/ServiceManagerBase.sol [hunk 5]
```diff
@@ -107,6 +107,39 @@ abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
         return restakedStrategies;
     }
 
+    function getQuorumsOfOperator(address _operator) external view returns (uint32[] memory){
+        bytes32 operatorId = _registryCoordinator.getOperatorId(_operator);
+        uint192 operatorQuorumBitmap = _registryCoordinator.getCurrentQuorumBitmap(operatorId);
+        return bytesArrayToUint32Array(BitmapUtils.bitmapToBytesArray(operatorQuorumBitmap));
+
```
Comment (gpsanant) [inline src/ServiceManagerBase.sol:114]: empty space

File: src/ServiceManagerBase.sol [hunk 6]
```diff
@@ -107,6 +107,39 @@ abstract contract ServiceManagerBase is IServiceManager, OwnableUpgradeable {
         return restakedStrategies;
     }
 
+    function getQuorumsOfOperator(address _operator) external view returns (uint32[] memory){
```
Comment (ypatil12) [inline src/ServiceManagerBase.sol:110]: We should also update the `ServiceManagerRouter`

Comment (ypatil12) [reply src/ServiceManagerBase.sol:110]: Enables subgraph to make calls to these functions

Comment (ypatil12) [reply src/ServiceManagerBase.sol:110]: (without having so many reverts and unreadable log files)

**Pull request closed without merging**